tests/zfs-tests/tests/functional/cli_root/zfs_snapshot/zfs_snapshot_009_pos.ksh

```diff
@@ -90,9 +90,11 @@ for i in 1 2 3; do
 done
 log_note "verify snapshot contents"
 for ds in $datasets; do
-	status=$(dircmp /$ds /$ds/.zfs/snapshot/snap | grep "different")
-	[[ -z $status ]] || log_fail "snapshot contents are different from" \
-	    "the filesystem"
+	diff $ds $ds/.zfs/snapshot/snap > /dev/null 2>&1
```

Comment:
I think you want `diff -r` here and elsewhere. Also in this diff line, I think you'll want a leading slash in front of `$ds`